internal/pkg/agent/cmd/otel.go

```diff
@@ -0,0 +1,123 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package cmd
+
+import (
+	"context"
+	"sync"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"github.com/elastic/elastic-agent-libs/service"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/errors"
+	"github.com/elastic/elastic-agent/internal/pkg/cli"
+	"github.com/elastic/elastic-agent/internal/pkg/otel"
+)
+
+const (
+	configFlagName = "config"
+	setFlagName    = "set"
+)
+
+func newOtelCommandWithArgs(_ []string, _ *cli.IOStreams) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "otel",
+		Short: "Start the Elastic Agent in otel mode",
+		Long:  "This command starts the Elastic Agent in otel mode.",
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			cfgFiles, err := getConfigFiles(cmd)
+			if err != nil {
+				return err
+			}
+			return runCollector(cmd.Context(), cfgFiles)
+		},
+		PreRun: func(c *cobra.Command, args []string) {
+			c.InheritedFlags().VisitAll(func(f *pflag.Flag) {
+				f.Hidden = true
+			})
+		},
+		SilenceUsage:  true,
+		SilenceErrors: true,
+	}
+
+	cmd.SetHelpFunc(func(c *cobra.Command, s []string) {
+		c.InheritedFlags().VisitAll(func(f *pflag.Flag) {
+			f.Hidden = true
+		})
```

Comment:
Nit: repeated in `PreRun` above; consider extracting into common function, assuming the reason for this hiding behavior is the same in both places.

Reply:
Also, would be good to add a comment as to why we're hiding inherited flags.